src/components/GoodsList/GoodsList.jsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export class GoodsList extends React.Component {
+  state = {
+    isReversed: false,
+    sortBy: '',
+  }
+
+  reverse = () => {
+    this.setState(prevState => ({
+      isReversed: !prevState.reverse,
+    }));
+  }
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortBy: '',
+    });
+  }
+
+  render() {
+    const { goods } = this.props;
+
+    goods.sort((a, b) => {
```

Comment:
```suggestion
    const { goods } = this.props;
    
    const filteredGoods = [...goods]

    filteredGoods.sort((a, b) => {
```
Make a copy of the incoming array from props and do everything with it